main.js

```diff
@@ -280,364 +716,113 @@ function render2D(foldObj) {
-
-        let sphCoords = new THREE.Spherical(rotationRadius, phi, theta);
-        camera.position.setFromSpherical(sphCoords);
-
-        // offset spherical position to new origin as well for panning
-        camera.position.addVectors(camera.position, origin);
-        camera.lookAt(origin);
-
-        renderer.render(scene, camera);
+    function foldAxiom1() {
+        createNewEdge(mathVertices[selectedPoints[0]], mathVertices[selectedPoints[1]]);
     }
 
-    function createFaceGeom(vertex_coords, faces_vertices) {
-        // both have to exist in order for the shape to be rendered
-        if (vertex_coords && faces_vertices) {
-            let triangleGeometry = new THREE.BufferGeometry();
-
-            if (vertex_coords.some(el => el.length !== 3)) {
-                vertex_coords = enforce3DCoordinates(vertex_coords);
+    function foldAxiom2() {
+        let bisector = new THREE.Vector3();
+        let lineBetween = new THREE.Line3(mathVertices[selectedPoints[0]],
+            mathVertices[selectedPoints[1]]);
+        let pointDirection = new THREE.Vector3();
+        lineBetween.delta(pointDirection);
+        let origin = new THREE.Vector3();
+        lineBetween.getCenter(origin);
+        bisector.crossVectors(pointDirection, new THREE.Vector3(0, 0, 1));
+
+        let intersections = lineAllCollisions(origin, bisector);
+        let mindex = 0;
+        let maxdex = 0;
+        for (let i = 0; i < intersections.length; i++) {
+            if (intersections[i][0] < intersections[mindex][0]) {
+                mindex = i;
+            } else if (intersections[i][1] < intersections[mindex][1]) {
+                mindex = i;
             }
-
-            // concatenating an empty array is a bit of a hack, bascially just squish the array
-            let vertsArray = deepArrayConcat(new Array(), vertex_coords);
-            vertsArray = mathCoordConversion(vertsArray);
-            const vertices = new Float32Array(vertsArray);
-
-            // check if there are any faces of more than three vertices
-            if (faces_vertices.some(el => el.length > 3)) {
-                faces_vertices = polygonToTri(faces_vertices);
+            if (intersections[i][0] > intersections[maxdex][0]) {
+                maxdex = i;
+            } else if (intersections[i][1] > intersections[maxdex][1]) {
+                maxdex = i;
             }
-            triangleGeometry.setIndex(deepArrayConcat(new Array(), faces_vertices));
-            triangleGeometry.setAttribute("position", new THREE.BufferAttribute(vertices, 3));
-
-            let plane = new THREE.Mesh(triangleGeometry,
-                new THREE.MeshBasicMaterial({color: 0x885556, side: THREE.DoubleSide}));
-            return plane;
-        } else {
-            alert("A shape was missing necessary information to be rendered");
-            return null;
         }
-    }
 
+        createNewEdge(new THREE.Vector3(intersections[mindex][0], intersections[mindex][1], 0),
```

Comment:
You're going to want to be a bit careful here — the intersection points might duplicate points that you already have. Imagine, for instance, folding a square in half parallel to an edge in both directions (to get four smaller squares and the intersection point in the center) and then folding along a diagonal; you'll get a duplicate copy of the center vertex.

Reply:
This is not immediately apparent, but `createNewEdge` calls `createNewVert` at every point it detects line intersections, and `createNewVert` only adds new vertices if and equivalent one does not exist already.